Shake/Main.lean

```diff
@@ -628,3 +632,11 @@ def main (args : List String) : IO UInt32 := do
   else
     println! "No edits required."
   return 0
+
+-- self-test so that future grammar changes cause a build failure
+/-- info: #[`Lake.CLI.Main] -/
+#guard_msgs (whitespace := lax) in
+#eval show MetaM _ from do
+  let (_, _, header, _) ← parseHeaderFromString (← getFileMap).source (← getFileName)
+  let imports := header.raw[2].getArgs
+  return imports.map fun i => i[3].getId
```

Comment:
I think this `i[3].getId` should be a helper function, so that we can't fix the test without fixing the code.